subprojects/core/src/main/java/org/gradle/internal/classpath/intercept/CallInterceptor.java

```diff
@@ -33,24 +29,7 @@
  * <p>
  * Descendants of this class should be thread-safe, making a stateless implementation is perfect.
  */
-public abstract class CallInterceptor implements BytecodeInterceptor {
-    private static final MethodHandles.Lookup LOOKUP = MethodHandles.lookup();
-
-    private static final MethodHandle INTERCEPTOR;
-
-    static {
-        try {
-            INTERCEPTOR = LOOKUP.findVirtual(CallInterceptor.class, "interceptMethodHandle", MethodType.methodType(Object.class, MethodHandle.class, int.class, String.class, Object[].class));
-        } catch (NoSuchMethodException | IllegalAccessException e) {
-            throw new GradleException("Failed to set up an interceptor method", e);
-        }
-    }
-
-    private final InterceptScope[] interceptScopes;
-
-    protected CallInterceptor(InterceptScope... interceptScopes) {
-        this.interceptScopes = interceptScopes;
-    }
+public interface CallInterceptor {
```

Comment:
🤔  This is called a "call" interceptor, but it intercepts `Invocation`s. Would there be a point in reconciling those terms?

Reply:
I think it's called CallInterceptor because it intercepts CallSites. I think name is fine